File: backend/app/schemas/action.py
from pydantic import BaseModel


class Action(BaseModel):
    id: str | None = None
    action_id: str | None = None

    device_id: int | None = None
    device: str | None = None

    status_id: int | None = None
    status: str | None = None

    performed_by_id: int | None = None
    performed_by: str | None = None

    assigned_to_id: int | None = None
    assigned_to: str | None = None

    shift_to_be_done_id: int | None = None
    shift_to_be_done: str | None = None

    date: str | None = None
    description: str | None = None
    last_updated: str | None = None

File: frontend/src/types/action.ts
export type Action = {
  id: string | null
  action_id: string | null

  device_id: number | null
  device: string | null

  status_id: number | null
  status: string | null

  performed_by_id: number | null
  performed_by: string | null

  assigned_to_id: number | null
  assigned_to: string | null

  shift_to_be_done_id: number | null
  shift_to_be_done: string | null

  date: string | null
  description: string | null
  last_updated: string | null
}